scripts/proteinInteractionEBI/parse_ebi_test.py

```diff
@@ -0,0 +1,123 @@
+    return file_terms
+
+CONST_FILE_TERMS = get_file_terms(TEST_TEXT)
+
+CONST_INTERACTION_TYPE_ID_SET = set(['MI:0045', 'MI:0401'])
+
+class TestParseEbi(unittest.TestCase):
+    """Test the functions in parse_ebi.py"""
+    def test_get_id_maps(self):
+        """Test function get_id_maps. Note that id_to_node here doesn't have parent_child
+        relation, so only map keys are tested."""
+        id_to_class_name, id_to_node = parse_ebi.get_id_maps(CONST_FILE_TERMS)
+        self.assertEqual(id_to_class_name, CONST_ID_TO_CLASS_NAME)
+        self.assertEqual(id_to_node.keys(), CONST_ID_TO_NODE_NO_RELATION.keys())
+    def test_build_child_parent_link(self):
+        """Test function build_child_parent_link by check the values of child_list and parent_list."""
+        id_to_node = copy.deepcopy(CONST_ID_TO_NODE_NO_RELATION)
+        id_to_node = parse_ebi.build_child_parent_link(CONST_FILE_TERMS, id_to_node)
+        def get_node_value_set(node_list):
+            value_set = set()
+            for node in node_list:
+                value_set.add(node.value)
+            return value_set
+        for id_key in id_to_node:
+            parent_value_set = get_node_value_set(id_to_node[id_key].parent_list)
+            const_parent_value_set = get_node_value_set(CONST_ID_TO_NODE[id_key].parent_list)
+            child_value_set = get_node_value_set(id_to_node[id_key].child_list)
+            const_child_value_set = get_node_value_set(CONST_ID_TO_NODE[id_key].child_list)
+            self.assertEqual(parent_value_set, const_parent_value_set)
+            self.assertEqual(child_value_set, const_child_value_set)
+    def test_TreeBuilder(self):
+        """Test TreeBuilder class."""
+        dfs_caller = parse_ebi.TreeBuilder(CONST_ID_TO_NODE)
+        INTERACTION_TYPE_ROOT = 'MI:0001'
+        interaction_type_id_set = dfs_caller.get_subset_id(INTERACTION_TYPE_ROOT)
+        self.assertEqual(interaction_type_id_set, CONST_INTERACTION_TYPE_ID_SET)
+
+    def test_get_schema_from_text(self):
+        """Test function get_schema_from_text by comparing the final schema."""
+        new_source_map = {'references':{}}
+        term = CONST_FILE_TERMS[1]
+        schema_res = parse_ebi.get_schema_from_text(term, CONST_ID_TO_NODE,
+                                                    new_source_map, CONST_ID_TO_CLASS_NAME,
+                                                    CONST_INTERACTION_TYPE_ID_SET, set(), set())
+        self.assertEqual(schema_res[0],
+                         ('Node: dcid:ExperimentalInteractionDetection\n'
+                          'typeOf: dcs:InteractionTypeEnum\n'
+                          'name: "ExperimentalInteractionDetection"\npsimiID: "MI:0045"\n'
+                          'description: "Methods base"\npubMedID: "14755292"\n'
+                          'specializationOf: dcs:InteractionDetectionMethod\n'
+                          'descriptionUrl: "http://psidev.info/groups/controlled-vocabularies"'))
```

Comment:
Make these as CONSTANTS enclosed in '''...'''?

Reply:
Fixed.